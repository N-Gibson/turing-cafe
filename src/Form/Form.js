import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      date: '',
      time: '',
      number: '',
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name] : e.target.value})
  }

  makeRes = () => {
    if(this.state.name !== '' && this.state.date !== '' && this.state.time !== '' && this.state.number !== '') {
      this.props.makeReservation(this.state)
      this.clearInputs();
    }
  }

  clearInputs = () => {
    this.setState({name: '', date: '', time: '', number: ''})
  }

  render() {
    return (
      <div className='form'>
        <input 
          type='text' 
          placeholder='Name'
          name='name'
          value={this.state.name}
          onChange={this.handleChange}
        ></input>
        <input 
          type='text' 
          placeholder='Date (mm/dd)'
          name='date'
          value={this.state.date}
          onChange={this.handleChange}
        ></input>
        <input 
          type='text' 
          placeholder='Time'
          name='time'
          value={this.state.time}
          onChange={this.handleChange}
        ></input>
        <input 
          type='text' 
          placeholder='Number of Guests'
          name='number'
          value={this.state.number}
          onChange={this.handleChange}
        ></input>
        <button type='button' onClick={this.makeRes}>Make Reservation</button>
      </div>
    )
  }
}
export default Form;