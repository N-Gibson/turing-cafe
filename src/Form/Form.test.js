import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper, makeRes;
  beforeEach(() => {
    makeRes = jest.fn()
    wrapper = shallow(<Form makeReservation={makeRes}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should be able to handle a change in state', () => {
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Bob'
      }
    }
    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('name')).toEqual('Bob')
  });

  it('should make a reservation on click', () => {
    wrapper.find('button').simulate('click')

    expect(makeRes.mock.calls.length).toEqual(1)
    // expect(makeRes).toHaveBeenCalled()
  })
})