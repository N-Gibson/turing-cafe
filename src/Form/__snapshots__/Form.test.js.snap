// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<div
  className="form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</div>
`;
